time-series/time-series-api/src/main/java/com/powsybl/timeseries/ast/NodeCalc.java

```diff
@@ -14,17 +14,19 @@
 
 import java.io.IOException;
 import java.io.UncheckedIOException;
-import java.util.List;
+import java.util.Deque;
 import java.util.Objects;
 
 /**
  * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
  */
 public interface NodeCalc {
 
-    <R, A> List<NodeCalc> acceptIterate(NodeCalcVisitor<R, A> visitor, A arg);
+    <R, A> R accept(NodeCalcVisitor<R, A> visitor, A arg, int depth);
```

Comment:
That would be the best place to explain the difference between accept, acceptIterate and acceptHandle.